src/LinkedList/linked_list.c

```diff
@@ -42,8 +46,16 @@ unsigned int appendToList(pLinkedList linkedList, T data) {
 
   // Because we are definitely appending a Node, lets make it.
   pNode newNode = (pNode) malloc(sizeof(Node));
+  if (newNode == NULL) {
+    printf("Malloc failed\n");
+    return 0;
+  }
   newNode->next = NULL;
   newNode->data = (T*) malloc(sizeof(T));
+  if (newNode->data == NULL) {
+    printf("Malloc failed\n");
+    return 0;
```

Comment:
Zero here is good because we are returning an unsigned int, where 0 == fail, 1 == success. Above (in makeList) we are returning a pointer, that's why we want to explicitly return NULL.